Mathlib/RingTheory/PowerSeries/Basic.lean

```diff
@@ -2679,4 +2739,84 @@ theorem algebraMap_apply'' :
   rfl
 #align power_series.algebra_map_apply'' PowerSeries.algebraMap_apply''
 
+end Algebra
+
+section Trunc
+/-
+Lemmas in this section involve the coercion `R[X] → R⟦X⟧`, so they may only be stated in the case
+`R` is commutative. This is because the coercion is an `R`-algebra map.
+-/
+variable {R : Type*} [CommSemiring R]
+
+open Nat hiding pow_succ pow_zero
+open Polynomial BigOperators Finset Finset.Nat
+
+theorem trunc_trunc_of_le {n m} (f : R⟦X⟧) (hnm : n ≤ m := by rfl) :
+    trunc n ↑(trunc m f) = trunc n f := by
+  ext d
+  rw [coeff_trunc, coeff_trunc, coeff_coe]
+  split_ifs with h
+  · rw [coeff_trunc, if_pos <| lt_of_lt_of_le h hnm]
+  · rfl
+
+@[simp] theorem trunc_trunc {n} (f : R⟦X⟧) : trunc n ↑(trunc n f) = trunc n f :=
+  trunc_trunc_of_le f
+
+@[simp] theorem trunc_trunc_mul {n} (f g : R ⟦X⟧) :
+    trunc n ( (trunc n f) * g : R⟦X⟧ ) = trunc n ( f * g ) := by
+  ext m
+  rw [coeff_trunc, coeff_trunc]
+  split_ifs with h
+  · rw [coeff_mul, coeff_mul, sum_congr rfl]
+    intro _ hab
+    have ha := lt_of_le_of_lt (antidiagonal.fst_le hab) h
+    rw [coeff_coe, coeff_trunc, if_pos ha]
+  · rfl
+
+@[simp] theorem trunc_mul_trunc {n} (f g : R ⟦X⟧) :
+    trunc n ( f * (trunc n g) : R⟦X⟧ ) = trunc n ( f * g ) := by
```

Comment:
```suggestion
    trunc n (f * (trunc n g) : R⟦X⟧) = trunc n (f * g) := by
```

Reply:
Thank you again. It seem I do this a lot.